### PR TITLE
Remove the pc machine

### LINKED ISSUE #1953: Remove support for the x86 PC machine
**Which feature do you think can be improved?**

There is no need to support both the PC and Q35 machine types for x86.

**How can it be improved?**

Just remove the PC machine.

**Additional Information**

Less machine types will mean less testing needed.

### PR DESCRIPTION
**Pull request by marcel-apf**
Fixes #1953

Depends-on: github.com/kata-containers/tests#3586
Signed-off-by: Marcel Apfelbaum <marcel@redhat.com>

### CODE REVIEW

File: src/runtime/virtcontainers/qemu.go [hunk 1]
```diff
@@ -2155,8 +2155,6 @@ func genericBridges(number uint32, machineType string) []types.Bridge {
 	case QemuQ35:
 		// currently only pci bridges are supported
 		// qemu-2.10 will introduce pcie bridges
-		fallthrough
-	case QemuPC:
 		bt = types.PCI
```
Comment (devimc) [inline src/runtime/virtcontainers/qemu.go:2158]: I wonder if we should change this to pcie-pci-bridges... :thinking:

**Review by devimc — COMMENTED** [review]
thanks @marcel-apf - one question

File: src/runtime/virtcontainers/qemu.go (hunk 1 above)
Comment (devimc) [reply src/runtime/virtcontainers/qemu.go:2158]: ok, we can leave it as it is for now..

**Comment by devimc** [comment]
can you please raise a pull request in the tests repository (ptal https://github.com/kata-containers/tests/blob/120353c1d0e89f76fa0fb3f4b8964e82e8df3350/.ci/install_runtime.sh#L20  and https://github.com/kata-containers/tests/blob/120353c1d0e89f76fa0fb3f4b8964e82e8df3350/Makefile#L168-L171) and add `Depends-on: github.com/kata-containers/tests#NNN` in this commit message to make sure we are testing q35 ?

**Comment by marcel-apf** [comment]
> can you please raise a pull request in the tests repository (ptal https://github.com/kata-containers/tests/blob/120353c1d0e89f76fa0fb3f4b8964e82e8df3350/.ci/install_runtime.sh#L20 and https://github.com/kata-containers/tests/blob/120353c1d0e89f76fa0fb3f4b8964e82e8df3350/Makefile#L168-L171) and add `Depends-on: github.com/kata-containers/tests#NNN` in this commit message to make sure we are testing q35 ?

@devimc thank you for pointing me to the tests repository, I'll be sure to make all the required changes.

**Comment by marcel-apf** [comment]
/cc @dgibson

> > can you please raise a pull request in the tests repository (ptal https://github.com/kata-containers/tests/blob/120353c1d0e89f76fa0fb3f4b8964e82e8df3350/.ci/install_runtime.sh#L20 and https://github.com/kata-containers/tests/blob/120353c1d0e89f76fa0fb3f4b8964e82e8df3350/Makefile#L168-L171) and add `Depends-on: github.com/kata-containers/tests#NNN` in this commit message to make sure we are testing q35 ?
> 
> @devimc thank you for pointing me to the tests repository, I'll be sure to make all the required changes.

@devimc done.
Please let me know if I missed anything

**Comment by devimc** [comment]
@marcel-apf thanks, sorry but you forgot `github.com/` in `Depends-on:` (`Depends-on: github.com/kata-containers/tests#3586`) (yes I know, that's not needed, I didn't implement it)

**Comment by marcel-apf** [comment]
> @marcel-apf thanks, sorry but you forgot `github.com/` in `Depends-on:` (`Depends-on: github.com/kata-containers/tests#3586`) (yes I know, that's not needed, I didn't implement it)

Done (I previously used the whole path:  https://... )

File: tools/agent-ctl/Cargo.lock [hunk 2]
```diff
@@ -985,7 +985,6 @@ dependencies = [
  "serde_json",
  "slog",
  "slog-scope",
- "tempfile",
```
Comment (dgibson) [inline tools/agent-ctl/Cargo.lock:988]: As discussed on our call, the Cargo.lock update is probably due to an earlier change forgetting to check in the Cargo.lock update.  I'd therefore split this out as a separate patch.

**Review by dgibson — COMMENTED** [review]
Looks good to me as far as it goes, but I think there are some more places to update:

- References to "pc" machine type in some docs, e.g. `docs/use-cases/Nvidia-GPU-passthrough-and-Kata.md` and the comments  in `src/runtime/cli/config/configuration-*.toml.in`.
- Simplifying logic in some places which don't directly reference QemuPC but do so implicitly, e.g. `genericAppendBridges`

**Comment by marcel-apf** [comment]
> Looks good to me as far as it goes, but I think there are some more places to update:
> 
> * References to "pc" machine type in some docs, e.g. `docs/use-cases/Nvidia-GPU-passthrough-and-Kata.md` and the comments  in `src/runtime/cli/config/configuration-*.toml.in`.

Sure, thanks for pointing me to to the files.

> * Simplifying logic in some places which don't directly reference QemuPC but do so implicitly, e.g. `genericAppendBridges`

I would prefer to simplify the logic in a different PR/issue to keep the patch set simple since the change is "disruptive" enough.


Thanks,
Marcel

**Comment by dgibson** [comment]
> > * Simplifying logic in some places which don't directly reference QemuPC but do so implicitly, e.g. `genericAppendBridges`
> 
> I would prefer to simplify the logic in a different PR/issue to keep the patch set simple since the change is "disruptive" enough.

Good point.  Agreed.

**Review by jodh-intel — APPROVED** [review]
Thanks @marcel-apf.

lgtm

File: tools/agent-ctl/Cargo.lock (hunk 2 above)
Comment (marcel-apf) [reply tools/agent-ctl/Cargo.lock]: @dgibson I rebased and this disappeared, however other diff appeared, I just left it out of the PR

**Comment by marcel-apf** [comment]
> > Looks good to me as far as it goes, but I think there are some more places to update:
> > 
> > * References to "pc" machine type in some docs, e.g. `docs/use-cases/Nvidia-GPU-passthrough-and-Kata.md` and the comments  in `src/runtime/cli/config/configuration-*.toml.in`.
> 
> Sure, thanks for pointing me to to the files.
> 

Done

Thanks,
Marcel

**Comment by marcel-apf** [comment]
> /test

It seems like I need someone that can run that command, @devimc @dgibson can you please ? :) 

Thanks,
Marcel

**Comment by devimc** [comment]
@marcel-apf `cri-containerd` test are failing

```
time="2021-06-16T11:35:35Z" level=fatal msg="run pod sandbox failed: rpc error: code = Unknown desc = failed to create containerd task: Add 13964MB virtio-mem-pci fail QMP command failed: Bus 'pcie.0' does not support hotplugging: unknown"
```

**Comment by marcel-apf** [comment]
> @marcel-apf `cri-containerd` test are failing
> 
> ```
> time="2021-06-16T11:35:35Z" level=fatal msg="run pod sandbox failed: rpc error: code = Unknown desc = failed to create containerd task: Add 13964MB virtio-mem-pci fail QMP command failed: Bus 'pcie.0' does not support hotplugging: unknown"
> ```

It seems we need to add to the CI configuration file an option to hotplug the virtio-mem-pci into the right place (pcie-root-port or the pcie-pci-bridge slot).
I will start looking into it.

**Comment by dgibson** [comment]
> > @marcel-apf `cri-containerd` test are failing
> > ```
> > time="2021-06-16T11:35:35Z" level=fatal msg="run pod sandbox failed: rpc error: code = Unknown desc = failed to create containerd task: Add 13964MB virtio-mem-pci fail QMP command failed: Bus 'pcie.0' does not support hotplugging: unknown"
> > ```
> 
> It seems we need to add to the CI configuration file an option to hotplug the virtio-mem-pci into the right place (pcie-root-port or the pcie-pci-bridge slot).
> I will start looking into it.

Oof.  I have a bad feeling about this.  I suspect this means that the virtio-mem stuff is hand-rolling its hotplug things, assumes its plugging on the root bus and won't work on q35 at all :/.

**Comment by marcel-apf** [comment]
Added an WIP patch that shows how https://github.com/kata-containers/govmm/pull/177 will be used.
I will updated the PR as soon as the other one gets merged.

**Comment by marcel-apf** [comment]
The second patch is not WIP anymore since https://github.com/kata-containers/govmm/pull/177 was merged.

**Comment by marcel-apf** [comment]
@devimc @fidencio Hi!
I have fixed the previous CI issue regarding the virtio-mem failure, the only error I see now in CI is:
```14:52:25 time="2021-06-22T11:52:25Z" level=debug msg=" Got result [[53 52 57 54 57 10]]"
14:52:25 time="2021-06-22T11:52:25Z" level=debug msg=" and got output [[54969]]"
14:52:25 time="2021-06-22T11:52:25Z" level=debug msg=" Iters is 1"
14:52:25 time="2021-06-22T11:52:25Z" level=debug msg=" Min is 54969.000000"
14:52:25 time="2021-06-22T11:52:25Z" level=debug msg=" Max is 54969.000000"
14:52:25 time="2021-06-22T11:52:25Z" level=debug msg=" Mean is 54969.000000"
14:52:25 time="2021-06-22T11:52:25Z" level=debug msg=" SD is 0.000000"
14:52:25 time="2021-06-22T11:52:25Z" level=debug msg=" CoV is 0.00"
14:52:25 time="2021-06-22T11:52:25Z" level=debug msg="Compare check for [blogbench]"
14:52:25 time="2021-06-22T11:52:25Z" level=debug msg="Checking value [mean]"
14:52:25 time="2021-06-22T11:52:25Z" level=debug msg=" Check minval (50227.794000 > 54969.000000)"
14:52:25 time="2021-06-22T11:52:25Z" level=debug msg=Passed
14:52:25 time="2021-06-22T11:52:25Z" level=debug msg=" Check maxval (61389.526000 < 54969.000000)"
14:52:25 time="2021-06-22T11:52:25Z" level=debug msg=Passed
14:52:25 time="2021-06-22T11:52:25Z" level=debug msg="Check for [blogbench] passed"
14:52:25 time="2021-06-22T11:52:25Z" level=debug msg=" with [[P blogbench 90.0% 98.5% 110.0% 20.0% 98.5% 98.5% 0.0% 0.0% 1]]"
14:52:25 time="2021-06-22T11:52:25Z" level=debug msg="Done blogbench"
14:52:25 time="2021-06-22T11:52:25Z" level=warning msg="Overall we failed"
14:52:25 
14:52:25 Report Summary:
14:52:25 +-----+----------------------+-------+--------+--------+-------+--------+--------+-------+-------+-----+
14:52:25 | P/F |         NAME         |  FLR  |  MEAN  |  CEIL  |  GAP  |  MIN   |  MAX   |  RNG  |  COV  | ITS |
14:52:25 +-----+----------------------+-------+--------+--------+-------+--------+--------+-------+-------+-----+
14:52:25 | *F* | boot-times           | 95.0% | 92.0%  | 105.0% | 10.0% | 83.3%  | 130.9% | 57.1% | 10.9% |  20 |
14:52:25 | P   | memory-footprint     | 95.0% | 101.5% | 105.0% | 10.0% | 101.5% | 101.5% | 0.0%  | 0.0%  |   1 |
14:52:25 | P   | memory-footprint-ksm | 95.0% | 101.5% | 105.0% | 10.0% | 101.5% | 101.5% | 0.0%  | 0.0%  |   1 |
14:52:25 | P   | blogbench            | 90.0% | 95.4%  | 110.0% | 20.0% | 95.4%  | 95.4%  | 0.0%  | 0.0%  |   1 |
14:52:25 | P   | blogbench            | 90.0% | 98.5%  | 110.0% | 20.0% | 98.5%  | 98.5%  | 0.0%  | 0.0%  |   1 |
14:52:25 +-----+----------------------+-------+--------+--------+-------+--------+--------+-------+-------+-----+
14:52:25 Fails: 1, Passes 4
14:52:25 Failed```

Is this a concern or the PR is in good shape?
Thanks,
Marcel

**Comment by devimc** [comment]
@marcel-apf this looks good to me, just need to [update the boot time baseline](https://github.com/kata-containers/tests/pull/3586#issuecomment-858698054)

```
06:52:25 time="2021-06-22T11:52:25Z" level=debug msg="Compare check for [boot-times]"
06:52:25 time="2021-06-22T11:52:25Z" level=debug msg="Checking value [mean]"
06:52:25 time="2021-06-22T11:52:25Z" level=debug msg=" Check minval (0.541500 > 0.524650)"
06:52:25 time="2021-06-22T11:52:25Z" level=warning msg="Failed Minval (0.541500 > 0.524650) for [boot-times]"
06:52:25 time="2021-06-22T11:52:25Z" level=debug msg=" Check maxval (0.598500 < 0.524650)"
```

**Comment by marcel-apf** [comment]
> @marcel-apf this looks good to me, just need to [update the boot time baseline](https://github.com/kata-containers/tests/pull/3586#issuecomment-858698054)
> 
> ```
> 06:52:25 time="2021-06-22T11:52:25Z" level=debug msg="Compare check for [boot-times]"
> 06:52:25 time="2021-06-22T11:52:25Z" level=debug msg="Checking value [mean]"
> 06:52:25 time="2021-06-22T11:52:25Z" level=debug msg=" Check minval (0.541500 > 0.524650)"
> 06:52:25 time="2021-06-22T11:52:25Z" level=warning msg="Failed Minval (0.541500 > 0.524650) for [boot-times]"
> 06:52:25 time="2021-06-22T11:52:25Z" level=debug msg=" Check maxval (0.598500 < 0.524650)"
> ```

Thanks @devimc I actually did that in a prev patch, however https://github.com/kata-containers/tests/commit/1a518b297f6a29cd3745b9464c9bb144b69e3780 updated the test to a lower value than mine. Now is even lower... I'll be sure to add that patch again.

**Comment by marcel-apf** [comment]
> > @marcel-apf this looks good to me, just need to [update the boot time baseline](https://github.com/kata-containers/tests/pull/3586#issuecomment-858698054)
> > ```
> > 06:52:25 time="2021-06-22T11:52:25Z" level=debug msg="Compare check for [boot-times]"
> > 06:52:25 time="2021-06-22T11:52:25Z" level=debug msg="Checking value [mean]"
> > 06:52:25 time="2021-06-22T11:52:25Z" level=debug msg=" Check minval (0.541500 > 0.524650)"
> > 06:52:25 time="2021-06-22T11:52:25Z" level=warning msg="Failed Minval (0.541500 > 0.524650) for [boot-times]"
> > 06:52:25 time="2021-06-22T11:52:25Z" level=debug msg=" Check maxval (0.598500 < 0.524650)"
> > ```
> 
> Thanks @devimc I actually did that in a prev patch, however [kata-containers/tests@1a518b2](https://github.com/kata-containers/tests/commit/1a518b297f6a29cd3745b9464c9bb144b69e3780) updated the test to a lower value than mine. Now is even lower... I'll be sure to add that patch again.

@devimc Done: https://github.com/kata-containers/tests/pull/3586/commits/467e333eeea7cfc98d71f73884dd9026adb0bff8
A "/test" command would be appreciated since I cannot do it myself yet.

Thanks,
Marcel

**Comment by marcel-apf** [comment]
Maybe we can re-run only the failed test?

**Review by devimc — APPROVED** [review]

**Review by fidencio — APPROVED** [review]
lgtm and I couldn't be happier to see some part of the code going away!

Thanks, @marcel-apf!

**Review by dgibson — APPROVED** [review]
Looks good.  Thanks Marcel.
